index.js

```diff
@@ -99,10 +100,12 @@ Choo.prototype.start = function () {
   var self = this
   if (this._historyEnabled) {
     this.emitter.prependListener(this._events.NAVIGATE, function () {
+      self.state.href = self._createLocation()
       self.state.query = nanoquery(window.location.search)
       if (self._loaded) {
-        self.emitter.emit(self._events.RENDER)
-        setTimeout(scrollToAnchor.bind(null, window.location.hash), 0)
+        var resList = self.emitter._listeners[self._events.RESOLVE]
+        var resCount = resList ? resList.length : 0;
```

Comment:
If we go with this implementation, it might be cleaner to move this logic to a `countListeners` or `hasListener` method in nanobus itself. But yeah, we'll cross that bridge if we get there.